docs/compatibility.md

```diff
@@ -64,6 +64,25 @@ is generally equivalent to
 client = httpx.Client(**kwargs)
 ```
 
+## Request instantiation
+
+Whenever you get a `Response` from an API call, the `request` attribute is actually the `Request` that was used internally. There are cases in which you want to build a `Request` object but don't want to send it immediately. Here is how you can do it.
+
+```python
+req = httpx.Request("GET", "https://example.org")
+```
+
+The above request object is ready to be sent. There’s no need for `prepare_request()` in HTTPX.
+
+```python
+with httpx.Client() as client:
+    r = client.send(req)
+```
+
+If you need to build a request while applying client defaults, see [here](https://www.python-httpx.org/advanced/#build-request).
+
+The list of arguments that are not present in `httpx.Request` class are `auth`, `timeout`, `allow_redirects`, `proxies`, `verify` and `cert` however these are available in `httpx.request`, `httpx.get`, `httpx.post` etc.
```

Comment:
Hi @lonewolf3739, thanks for updating the PR. :-)

Reading this bit I realized there were a few pieces of info that were missing entirely from our docs, and we could use as a prerequisite to make this section as focused as possible.

Basically once #836 gets in I believe we can strip this down to something like…

```suggestion
There is no notion of [prepared requests](https://requests.readthedocs.io/en/stable/user/advanced/#prepared-requests) in HTTPX. If you need to customize request instantiation, see [Request instances](/advanced#request-instances).

Besides, `httpx.Request()` does not support the `auth`, `timeout`, `allow_redirects`, `proxies`, `verify` and `cert` parameters. However these are available in `httpx.request`, `httpx.get`, `httpx.post` etc., as well as on [`Client` instances](/advanced#client-instances).
```

Reply:
Revamped doc here #836 looks great :+1: